src/store/reducers/subjects.js

```diff
@@ -35,14 +40,28 @@ const subjects = (state = { primary: [], lingual: [] }, action) => {
           subject => subject.id !== action.payload.id
         )
       }
-    case UPDATE_SUBJECT:
+    case UPDATE_SUBJECT_NAME:
       return {
         ...state,
         [action.payload.subjectsType]: state[action.payload.subjectsType].map(
           subject => (subject.id === action.payload.id)
             ? {
               ...subject,
-              [action.payload.whatsUpdated]: action.payload.newValue
+              name: action.payload.name,
```

Comment:
Why can't you just destructure everything you have in action.payload here? That way you would not need to have separate actions for updating.

BTW if you worry about having in payload some other properties (like `id` or `subjectsType`) you would not like to pass, consider using action structure having `meta` key.
```
{
  type: UPDATE_SUBJECT_NAME,
  payload: {
    name,
    // Whatever you want to update/destructure
  },
  meta: {
    id,
    subjectsType,
  }
}
```
That's the way we do in large projects to prevent micro management like this.

Reply:
What do you mean with 

> `Why can't you just destructure everything you have in action.payload here? That way you would not need to have separate actions for updating.`

?

Initially I had an universal update action creator and reducer. The first approach was to always pass name, elementaryScore and advancedScore and update everything (despite the fact only one could change). In the second approach I used the `['whatsupdated']: newValue` as it's seen before PR. But it seemed to be vague when being used in component so I decided to split the logic (despite having more code).

Reply:
Problem is, none of this solutions are scaling well when app grows even to the medium size.

What I suggest is 
reducer:
```
case UPDATE_SUBJECT:
  return {
    ...state,
    [action.meta.subjectsType]: state[action.meta.subjectsType].map(
      subject => (subject.id === action.meta.id)
        ? { ...subject, ...action.payload }
        : subject
    )
  }
```

actionCreator:
```
export const updateSubject = ({ id, subjectsType, ...updatedProperties }) => {
  if (updatedProperties.elementaryScore && updatedProperties.advancedScore) {
    updatedProperties.maxScore = getSubjectScore(updatedProperties.elementaryScore, updatedProperties.advancedScore)
  }
  return {
    type: UPDATE_SUBJECT,
    payload: updatedProperties,
    meta: { id, subjectsType }
  }
}
```

usage somewhere in components
```
this.props.updateSubject({id, subjectsType, name: 'new name'})
this.props.updateSubject({id, subjectsType, elementaryScore: 40, advancedScore: 20})
```